tools/demo-library/generate.py

```
@@ -0,0 +1,201 @@
+                           ffprobe_duration(cand) >= ffprobe_duration(raw):
+                            cand.replace(raw)
+                time.sleep(4)
+            if raw.exists() and ffprobe_duration(raw) >= ACCEPT_SECONDS:
+                tag(raw, final, cover if cover.exists() else None,
+                    {**tr, "artist": album["artist"], "album": album["album"],
+                     "year": album["year"], "genre": album["genre"], "total": total})
+                print(f"    ok {ffprobe_duration(final):.0f}s")
+            else:
+                failures.append(f"track: {album['album']} / {tr['title']}")
+
+    print("\nDone." if not failures else f"\nDone with failures:\n  " + "\n  ".join(failures))
```

Ruff (0.16.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)